frontend/src/hooks/useTerminal.ts

```diff
@@ -131,10 +134,17 @@ export const useTerminal = (commands: Command[] = []) => {
     if (terminal.current && commands.length > 0) {
       // Start writing commands from the last command index
       for (let i = lastCommandIndex.current; i < commands.length; i += 1) {
-        const command = commands[i];
-        terminal.current?.writeln(parseTerminalOutput(command.content));
+        // eslint-disable-next-line prefer-const
+        let { content, type } = commands[i];
 
-        if (command.type === "output") {
+        if (secrets.length > 0) {
+          const secretRegex = new RegExp(secrets.join("|"), "g");
```

Comment:
this is going to be very fallible, esp if `secrets` contains regex characters like `?` or `*`

I'd just do
```
secrets.forEach(secret => content = content.replaceAll(secret, '*****'))
```